clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -943,11 +943,35 @@ static LValue emitGlobalVarDeclLValue(CIRGenFunction &CGF, const Expr *E,
   return LV;
 }
 
-static LValue emitCapturedFieldLValue(CIRGenFunction &CGF, const FieldDecl *FD,
-                                      mlir::Value ThisValue) {
-  QualType TagType = CGF.getContext().getTagDeclType(FD->getParent());
-  LValue LV = CGF.MakeNaturalAlignAddrLValue(ThisValue, TagType);
-  return CGF.emitLValueForField(LV, FD);
+static LValue emitCapturedFieldLValue(CIRGenFunction &cgf, const FieldDecl *fd,
+                                      mlir::Value thisValue) {
+  return cgf.emitLValueForLambdaField(fd, thisValue);
+}
+
+/// Given that we are currently emitting a lambda, emit an l-value for
+/// one of its members.
+///
+LValue CIRGenFunction::emitLValueForLambdaField(const FieldDecl *field,
+                                                mlir::Value thisValue) {
+  bool hasExplicitObjectParameter = false;
+  const auto *methD = dyn_cast_if_present<CXXMethodDecl>(CurCodeDecl);
+  if (methD) {
+    hasExplicitObjectParameter = methD->isExplicitObjectMemberFunction();
+    assert(methD->getParent()->isLambda());
+    assert(methD->getParent() == field->getParent());
+  }
+  LValue lambdaLV;
+  if (hasExplicitObjectParameter) {
+    llvm_unreachable("ExplicitObjectMemberFunction NYI");
+  }
```

Comment:
Please put this logic under the `} else {` already

Reply:
This comment didn't get addressed.

Reply:
Sorry I misunderstood you. Are you suggesting 
we do sth like:
`if(!hasExplicitObjectParameter) {
   ...
} else {
   NYI
}
` ?
I didn't have an `else` on `NYI` because of [llvm has a coding style don-t-use-else-after-a-return](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return)
Also, in the same spirit, `NYI` seems to me like `return` in the sense of unreachable afterwards. So I didn't have `else` in the first version.
Please let me know if I understand it right.